datalad/distributed/create_sibling_gitlab.py

```diff
@@ -0,0 +1,598 @@
+    ignore_var = 'remote.{}.annex-ignore'.format(remotename)
+    if ignore_var not in ds.config:
+        ds.config.add(ignore_var, 'true', where='local')
+
+    for res in ds.siblings(
+            'configure',
+            name=remotename,
+            url=site_project['http_url_to_repo']
+            if access in ('http', 'ssh+http')
+            else site_project['ssh_url_to_repo'],
+            pushurl=site_project['ssh_url_to_repo']
+            if access in ('ssh', 'ssh+http')
+            else None,
+            recursive=False,
+            publish_depends=depends,
+            result_renderer='disabled',
+            return_type='generator'):
+        yield res
+
+
+class GitLabSite(object):
+    def __init__(self, site):
+        import gitlab
+        self.gitlab = gitlab
+        try:
+            self.site = gitlab.Gitlab.from_config(site)
+        except gitlab.config.GitlabDataError as e:
+            raise ValueError(
+                '{}, please configure access to this GitLab instance'.format(
+                    str(e)))
+
+    def get_project(self, path):
+        try:
+            return self.site.projects.get(path).attributes
+        except self.gitlab.GitlabGetError as e:
+            lgr.debug("Project with path '%s' does not yet exist at site '%s'",
+                      path, self.site.url)
+            return None
+
+    def create_project(self, path, description=None):
+        path_l = path.split('/')
+        namespace_id = self._obtain_namespace(path_l)
+        # check for options:
+        # https://gitlab.com/help/api/projects.md#create-project
+        props = dict(
+            name=path_l[-1],
+            namespace_id=namespace_id,
+        )
+        if description:
+            props['description'] = description
+        project = self.site.projects.create(props)
+        return project.attributes
+
+    def _obtain_namespace(self, path_l):
+
+        if len(path_l) == 1:
+            # no nesting whatsoever
+            return None
+
+        try:
+            namespace_id = self.site.groups.get(
+                '/'.join(path_l[:-1])).get_id()
+            lgr.debug("Found existing parent group '%s' with ID %s",
+                      '/'.join(path_l[:-1]), namespace_id)
+        except self.gitlab.GitlabGetError as e:
+            try:
+                if len(path_l) > 2:
+                    parent_group = self.site.groups.get(
+                        '/'.join(path_l[:-2]))
+                else:
+                    parent_group = None
+            except self.gitlab.GitlabGetError as e:
+                raise ValueError(
+                    "No parent group {} for project {} found, "
+                    "and a group {} also does not exist. At most one "
+                    "parent group would be created.".format(
+                        '/'.join(path_l[:-1]),
+                        '/'.join(path_l),
+                        '/'.join(path_l[:-2]),
+                    ))
```

Comment:
again the same too strong of an assumption on when GitlabGetError would be thrown...?